### PR TITLE
S23 allow various prefix for links (new commit on develop as base instead of recim)

### PR DESCRIPTION
**Pull request by mla04762**
#1820

### CODE REVIEW

**Comment by mla04762** [comment]
Need to fix some errors before pushing

**Comment by bennettforkner** [comment]
Instead of closing this PR, you can just convert it to a draft if you would like. It's a small button in the right-hand sidebar.

File: src/components/Profile/components/Identification/components/SocialMediaLinks/components/LinksDialog/index.jsx [hunk 1]
```diff
@@ -23,9 +23,17 @@ const LinksDialog = ({ links, createSnackbar, onClose, setLinks }) => {
...
-      value === '' || value.indexOf(prefix) === 0 || (prefix2 && value.indexOf(prefix2) === 0);
+    const { prefix } = socialMediaInfo[platform];
+    let isValid = value === '';
+    console.log('isValid' + isValid);
+    if (!isValid && prefix) {
+      for (let i = 0; i < prefix.length; i++) {
+        if (value.indexOf(prefix[i]) === 0) {
+          isValid = true;
+        }
+      }
```
Comment (bennettforkner) [inline src/components/Profile/components/Identification/components/SocialMediaLinks/components/LinksDialog/index.jsx:34]: FYI, I know this is a draft, but normal `for` loops are prone to errors, so we tend to prefer functions like `.map(`, `.filter(`, `.some(`, or `.foreach(`.

`.some(...)` checks if any of the elements in the collection return true for the logic in the lamda function.

For example:
```suggestion
      isValid = prefix.some(p =>
        value.indexOf(p) === 0);
```

**Comment by andrew-wzj** [comment]
The problem of making social media format without both https and www, it cause the page doesn't redirect to the social media page but search in 360 locally by changing url to localhost/facebook.com/....

**Comment by andrew-wzj** [comment]
The expected change here is to matching user input url with each valid prefix in the list. Later, the further improvement could be if user didn't type https or www or both, it would help user to save urls in right format.

**Review by jsenning — APPROVED** [review]
LGTM

**Pull request merged**